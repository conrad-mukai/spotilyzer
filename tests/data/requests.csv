pod-name,replicas,core-limit,mem-limit
web-app-a,100,2,4
spark-executor-a,50,4,24
spark-executor-b,100,3,9
web-app-b,50,2,5
spark-executor-c,50,4,28
web-app-c,200,2,8
spark-executor-d,100,2,16
