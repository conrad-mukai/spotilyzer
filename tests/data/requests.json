{
  "requests": [
    {
      "pod-name": "web-app-a",
      "replicas": 100,
      "core-limit": 2.0,
      "mem-limit": 4.0
    },
    {
      "pod-name": "spark-executor-a",
      "replicas": 50,
      "core-limit": 4.0,
      "mem-limit": 24.0
    },
    {
      "pod-name": "spark-executor-b",
      "replicas": 100,
      "core-limit": 3.0,
      "mem-limit": 9.0
    },
    {
      "pod-name": "web-app-b",
      "replicas": 50,
      "core-limit": 2.0,
      "mem-limit": 5.0
    },
    {
      "pod-name": "spark-executor-c",
      "replicas": 50,
      "core-limit": 4.0,
      "mem-limit": 28.0
    },
    {
      "pod-name": "web-app-c",
      "replicas": 200,
      "core-limit": 2.0,
      "mem-limit": 8.0
    },
    {
      "pod-name": "spark-executor-d",
      "replicas": 100,
      "core-limit": 2.0,
      "mem-limit": 16.0
    }
  ]
}