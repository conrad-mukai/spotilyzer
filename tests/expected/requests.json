{
  "requests": [
    {
      "pod-name": "web-app-a",
      "replicas": 100,
      "core-limit": 2.0,
      "mem-limit": 4.0,
      "group-name": "compute"
    },
    {
      "pod-name": "spark-executor-a",
      "replicas": 50,
      "core-limit": 4.0,
      "mem-limit": 24.0,
      "group-name": "memory"
    },
    {
      "pod-name": "spark-executor-b",
      "replicas": 100,
      "core-limit": 3.0,
      "mem-limit": 9.0,
      "group-name": "general"
    },
    {
      "pod-name": "web-app-b",
      "replicas": 50,
      "core-limit": 2.0,
      "mem-limit": 5.0,
      "group-name": "compute"
    },
    {
      "pod-name": "spark-executor-c",
      "replicas": 50,
      "core-limit": 4.0,
      "mem-limit": 28.0,
      "group-name": "memory"
    },
    {
      "pod-name": "web-app-c",
      "replicas": 200,
      "core-limit": 2.0,
      "mem-limit": 8.0,
      "group-name": "general"
    },
    {
      "pod-name": "spark-executor-d",
      "replicas": 100,
      "core-limit": 2.0,
      "mem-limit": 16.0,
      "group-name": "memory"
    }
  ]
}